src/components/SignupCard/index.jsx

```diff
@@ -1,36 +1,45 @@
 import React, { useState } from 'react';
 import { Link, useHistory } from 'react-router-dom';
 
-import { Button, Card, Form, Input, Space, Typography } from 'antd';
+import { Button, Card, Form, Input, Modal, Space, Typography } from 'antd';
 import { LockOutlined, UserOutlined, MailOutlined } from '@ant-design/icons';
 
-import './SignupCard.scss';
-import { useAuth } from 'services/AuthProvider';
+import { useAuth } from 'hooks/Hooks';
+import routes from 'constants/routes';
+
+import './styles.scss';
 
 const { Text } = Typography;
 
 function SignupCard() {
   const { signup } = useAuth();
-  // eslint-disable-next-line no-unused-vars
-  const [loading, setLoading] = useState(false);
+  const [visible, setVisible] = useState(false);
+  const [modalText, setModalText] = useState('');
   const history = useHistory();
 
+  const promptModal = (message) => {
+    setVisible(true);
+    setModalText(message);
+  };
+
+  const handleCancel = () => {
+    setVisible(false);
+  };
+
   async function handleSubmit(userInput) {
     try {
-      /* CHANGE LOADER */
-      setLoading(true);
       await signup(userInput.email, userInput.password, userInput.username);
-      setLoading(false);
-      return history.push('/');
+      history.push(routes.ROOT);
     } catch (errors) {
-      /* FOR TESTING (NEEDS IMPLEMENTATION OF ERROR HANDLING) */
-      // eslint-disable-next-line no-console
-      return console.log(errors.message);
+      promptModal(errors.message);
     }
   }
 
   return (
     <Card className="Login-card">
+      <Modal title="Signup Failed" centered visible={visible} onCancel={handleCancel} footer={null}>
```

Comment:
same comment about sorting props

Reply:
do we really have to pass in `null` for footer?

Reply:
By default po sa antd documentation if footer is not specified to null it will have two footers - cancel and ok buttons